benchmark/pkg/scraper/scraper.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scraper
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"strings"
+
+	"k8s.io/contrib/test-utils/utils"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+
+	"github.com/golang/glog"
+)
+
+const (
+	readingJunk              = iota
+	readingAPICallLatencies  = iota
+	readingPodStartupLatency = iota
+	buildLogPath             = "build-log.txt"
+	testNameIdentifierTag    = "[BeforeEach] [k8s.io]"
+	apiLatencyResultTag	     = "API calls latencies: "
+	podLatencyResultTag	     = "Pod startup latency: "
+)
+
+// This function takes a job name and a run number, gets that run's build log,
+// reads it and parses JSON summaries of API call and Pod startup latencies.
+func GetMetricsForRun(job string, run int) (map[string]*e2e.APIResponsiveness, map[string]*e2e.PodStartupLatency, error) {
+	// Get the build log for the run.
+	googleGCSBucketUtils := utils.NewUtils(utils.KubekinsBucket, utils.LogDir)
+	response, err := googleGCSBucketUtils.GetFileFromJenkinsGoogleBucket(job, run, buildLogPath)
+	if err != nil {
+		return nil, nil, err
+	}
+	body := response.Body
+	defer body.Close()
+	scanner := bufio.NewScanner(body)
+
+	// Parse the log using a finite state automaton.
+	buff := &bytes.Buffer{}
+	apiCallLatencies := make(map[string]*e2e.APIResponsiveness)
+	podStartupLatency := make(map[string]*e2e.PodStartupLatency)
+	testName := ""
+	state := readingJunk
+	for scanner.Scan() {
```

Comment:
This is not really how state machines are implemented. It's more readable to switch on state, not on the line, as then you have clearly defined transitions.